src/marqo/core/structured_vespa_index/structured_vespa_index.py

```
@@ -615,6 +617,13 @@ def _to_vespa_hybrid_query(self, marqo_query: MarqoHybridQuery) -> Dict[str, Any
                         facet_queries.append(facets_query_skeleton % (query_yql, new_facets_term))
             facet_queries = QUERY_DELIMITER.join(facet_queries)
 
+        query_hash_without_offset = None
+        if marqo_query.offset % marqo_query.limit == 0:
+            if marqo_query.hybrid_parameters.retrievalMethod.lower() == RetrievalMethod.Disjunction:
+                query_hash_without_offset = marqo_query.get_query_hash_without_offset()
+        else:
+            logger.warning("Offset is not a multiple of limit. This is a wrong usage of pagination.")
```

I guess no one will see this warning. Should we just raise an error and block this usage?

---

Well, not really. It is okay to support it if there are some edge usecases, blocking it is a little extra. Warning is just for us to see if we are trying to understand if customer is doing something wrong.